controllers/kokumetricsconfig_controller.go

```diff
@@ -674,7 +689,21 @@ func (r *KokuMetricsConfigReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	}
 
 	// attempt to collect prometheus stats and create reports
-	collectPromStats(r, kmCfg, dirCfg)
+	if err := getPromCollector(r, kmCfg); err != nil {
+		log.Error(err, "failed to get prometheus connection")
+		return ctrl.Result{RequeueAfter: time.Minute * 2}, err // give things a break and try again in 2 minutes
```

Comment:
You could use exponential backoff here starting with a smaller value.

Reply:
Why?

Reply:
To potentially reduce overall wait time and/or avoid overwhelming the service.

It may not make sense in this particular scenario, but it's a good technique with decentralized clients interacting with a central service.

Not a blocker, just a comment. It's something that always comes to mind when I see fixed wait times.